xarray_jax/__init__.py

```diff
@@ -279,20 +273,14 @@ def assign_coords(
   coords = {} if coords is None else dict(coords)  # Copy before mutating.
```

Comment:
I don't think this line is needed any more either, since we don't mutate `coords` and you check for the None case below.